File: app/src/main/java/com/example/new1/storage/db/RoomContentDao.java
package com.example.new1.storage.db;

import androidx.annotation.NonNull;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface RoomContentDao {
    @Query("SELECT * FROM room_content WHERE storage_key = :storageKey LIMIT 1")
    RoomContentEntity findByStorageKey(@NonNull String storageKey);

    @Query("SELECT storage_key FROM room_content")
    List<String> listStorageKeys();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void upsert(@NonNull RoomContentEntity entity);

    @Query("DELETE FROM room_content WHERE storage_key = :storageKey")
    void deleteByStorageKey(@NonNull String storageKey);
}

File: app/src/main/java/com/example/new1/data/RoomContentDao.kt
package com.example.new1.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface RoomContentDao {
    @Query("SELECT * FROM room_content WHERE storage_key = :storageKey LIMIT 1")
    suspend fun findByStorageKey(storageKey: String): RoomContentEntity?

    @Query("SELECT storage_key FROM room_content")
    suspend fun listStorageKeys(): List<String>

    @Query("SELECT COUNT(*) FROM room_content")
    fun count(): Int

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsert(entity: RoomContentEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun upsertAllSync(entities: List<RoomContentEntity>)

    @Query("DELETE FROM room_content WHERE storage_key = :storageKey")
    suspend fun deleteByStorageKey(storageKey: String)
}
